add a RelatedGroups component to show on bottom of public page

asood123 opened the pull request.

xdamman commented on frontend/src/components/CollectiveCard.js:
```diff
@@ -29,11 +48,11 @@ export default class CollectiveCard extends Component {
             <div className='CollectiveCard-body'>
               <div className='CollectiveCard-name'>{name}</div>
-              <div className='CollectiveCard-description'>{description}</div>
+              <div className='CollectiveCard-description'>We are on a mission to {description}</div>
```
Now that this is not only for the homepage, we should move that to the `strings.json` and use `i18n`.

But it's tricky because 
- you should actually use the language of that group and not of that page when you render it.
- you should only show related collectives in the same language.

asood123 replied:
same language is too restrictive right now. Let's come back to it when we have a few hundred active collectives.

xdamman commented on frontend/src/containers/PublicGroup.js:
```diff
@@ -79,7 +80,11 @@ export class PublicGroup extends Component {
             newUserId={newUser.id}
             closeDonationModal={this._closeDonationFlow.bind(this)} />
+          <section className='pt4'>
+            <RelatedGroups title={'Check out similar collectives'} groupList={group.related} />
```
we need this in i18n, e.g. after you donate to PartidoDigital or LaPrimaire.org, you want to show text in spanish/french.

asood123 commented:
Found a bug. Don't merge yet.

asood123 commented:
Added a bunch of changes
- serverside rendering for homepage
- fixed an issue that was losing us most of the benefits of ssr
- added logic to not fetch data again after ssr (added a new reducer)
- few other login tweaks

xdamman commented on frontend/src/containers/HomePage.js:
```diff
@@ -79,7 +52,7 @@ export class HomePage extends Component {
         </section>
         <section className='HomePageInfo' id="howitworks">
-          <div className='heading'>What is a Open Collective?</div>
+          <div className='heading'>What is an Open Collective?</div>
```
I'd argue we should write "an open collective" (or "Open Collective" should refer to the company)

xdamman commented on server/routes.js:
```diff
@@ -55,15 +55,15 @@ module.exports = (app) => {
-  app.get('/:slug/banner.md', mw.cache(300), mw.fetchGroupBySlug, mw.fetchUsers(), controllers.banner.markdown);
-  app.get('/:slug/banner.js', mw.cache(3000), mw.fetchGroupBySlug, mw.fetchUsers(), controllers.banner.js);
-  app.get('/:slug/:tier.md', mw.cache(300), mw.fetchGroupBySlug, mw.fetchUsers(), controllers.banner.markdown);
+  app.get('/:slug/banner.md', mw.cache(300), controllers.profile, mw.fetchUsers(), controllers.banner.markdown);
```
this should be a middleware, you can only have one controller per route (by design)

xdamman commented on server/routes.js:
```diff
@@ -72,7 +72,7 @@ module.exports = (app) => {
    * the explicit routes and just do `app.use(render)`
    */
-  app.get('/', mw.ga, mw.addTitle('OpenCollective - Collect and disburse money transparently'), render);
+  app.get('/', mw.ga, controllers.homepage, mw.addTitle('OpenCollective - Collect and disburse money transparently'), render);
```
Middlewares, as their name suggests, should be in the middle. You should end with the controller.

xdamman commented on server/routes.js:
```diff
@@ -83,12 +83,12 @@ module.exports = (app) => {
   app.get('/connect/github', mw.ga, render);
   app.get('/:slug/connect/twitter', mw.ga, render);
-  app.get('/:slug/edit-twitter', mw.ga, mw.fetchGroupBySlug, render);
+  app.get('/:slug/edit-twitter', mw.ga, controllers.profile, render);
```
Ideally, the controller should call render.
(but it was already like that, it's ok to keep it that way for now if you want)

xdamman commented on server/views/pages/app.hbs:
```diff
@@ -22,7 +22,7 @@
     {{> sprite}}
 
-    <div id="content">{{{html}}}</div>
+    <div id="content"><div>{{{html}}}</div></div>
```
why this extra div?

asood123 replied:
a react artifact that was preventing efficient serverside rendering.
